src/lib/mina_base/party.ml

```diff
@@ -412,6 +480,48 @@ module Body = struct
       ; Random_oracle_input.field call_data
       ]
 
+  let gen ?pk () : t Quickcheck.Generator.t =
+    let open Quickcheck.Let_syntax in
+    let%bind pk =
+      match pk with Some pk -> return pk | None -> Public_key.Compressed.gen
+    in
+    let%bind update = Update.gen in
+    let%bind token_id = Token_id.gen in
+    let%bind delta =
+      let%bind magnitude = Currency.Amount.gen in
+      let%map sgn = Quickcheck.Generator.of_list [ Sgn.Pos; Neg ] in
+      Currency.Signed_poly.{ magnitude; sgn }
+    in
+    let field_array_list_gen ~max_array_len ~max_list_len =
+      let array_gen =
+        let%bind array_len = Int.gen_uniform_incl 0 max_array_len in
+        let%map fields =
+          Quickcheck.Generator.list_with_length array_len
+            Snark_params.Tick.Field.gen
+        in
+        Array.of_list fields
+      in
+      let%bind list_len = Int.gen_uniform_incl 0 max_list_len in
+      Quickcheck.Generator.list_with_length list_len array_gen
+    in
+    (* TODO: are these lengths reasonable? *)
+    let%bind events = field_array_list_gen ~max_array_len:8 ~max_list_len:12 in
+    let%bind rollup_events =
+      field_array_list_gen ~max_array_len:4 ~max_list_len:6
+    in
+    let%bind call_data = Snark_params.Tick.Field.gen in
+    (* TODO: are these numbers reasonable? *)
```

Comment:
`depth`s are supposed to increase by a maximum step of `1`, but can be decreased back by any amount as long as they stay positive. Perhaps there's a better representation for this concept that we can consider, such as
```ocaml
{ Party.data = { body = { child_parties = [...]; ... }; ... }
```
Wdyt?

Reply:
For ease, it might make sense to set them all as `0` here and do this in the `Parties.t` generator instead

Reply:
@mrmr1993 As you traverse the `other_parties` list, then, the depth in the list head is an arbitrary integer, and from a list element to the next, the depth may increase by one, stay the same, or be decreased to any value geq 1. Is that the invariant that must hold?